doc/sphinx/addendum/implicit-coercions.rst

```diff
@@ -167,25 +167,38 @@ Declaring Coercions
      Silence the non uniform inheritance warning.
 
   .. exn:: @qualid not declared.
-     :undocumented:
+
+     The given :token:`qualid` is not a defined constant.
 
   .. exn:: @qualid is already a coercion.
-     :undocumented:
+
+     The given :token:`qualid` is already registered a coercion.
 
   .. exn:: Funclass cannot be a source class.
-     :undocumented:
+
+     Funclass as a source class is not supported.
```

Comment:
Something like `f : (A -> B) -> (A -> B)` would be ambiguous to define as a coercion since for `g : A -> B`, `g a` could also mean `f g a`.

Reply:
That seems like a problem with coercions where source and target are the same, nothing about functions.

Reply:
OK here is a more complicated example: Make `f : (A -> B) >-> (A -> C)` a coercion. Take `g : A -> B`. Suppose there is a coercion `h : C >-> B`. Take the term `g a` it could also be `h (f g a)`.

Reply:
What does "it could also be" mean?
Do you mean it's harder to detect loops when there is source funclass?

Also do you have a source for this being why there is this limitation?

Reply:
It seems like maybe there is a genuine issue when the source and target classes are the same (in this case, if both are Funclass, even if the types are different).  Coercions from Funclass to things that are not Funclass seem fine, right?

Reply:
I've reworded to "currently not supported" and that it "may change in the future". I don't think it is worth saying anything more since there doesn't appear to be a concrete reason I can think of.